cmd/flags_core.go

```diff
@@ -38,54 +36,31 @@ func CoreFlags() *flag.FlagSet {
 		"9090",
 		"Set a custom gRPC port for the core node connection. The --core.ip flag must also be provided.",
 	)
-
 	return flags
 }
 
 // ParseCoreFlags parses Core flags from the given cmd and applies values to Env.
-func ParseCoreFlags(ctx context.Context, cmd *cobra.Command) (context.Context, error) {
+func ParseCoreFlags(
+	ctx context.Context,
+	cmd *cobra.Command,
+	cfg *nodebuilder.Config,
+) (setCtx context.Context, err error) {
+	defer func() {
+		setCtx = WithNodeConfig(ctx, cfg)
+	}()
```

Comment:
Looks like`setCtx` value  could also be set on return without defer func. I think there is no need for defering with clear return path.

On the side note it seems like some of the parsing flags functions are doing kinda same thing: they fill `nodebuilder.Config` struct and then transfer values to context. Maybe it is possible to only fill the Config in those parse functions and then move all values to ctx in the end once?

Reply:
> Looks likesetCtx value could also be set on return without defer func. I think there is no need for defering with clear return path.

Actually true. This would look much cleaner